monai/losses/unified_focal_loss.py

```
@@ -162,51 +241,53 @@ def __init__(
-            ValueError: When input and target are different shape
-            ValueError: When len(y_pred.shape) != 4 and len(y_pred.shape) != 5
-            ValueError: When num_classes
-            ValueError: When the number of classes entered does not match the expected number
+            y_pred: Prediction logits. Shape: (B, C, H, W, [D]).
+                    Supports binary (C=1 or C=2) and multi-class (C>2) segmentation.
+            y_true: Ground truth labels. Shape should match y_pred (or be indices if to_onehot_y is True).
         """
         if y_pred.shape != y_true.shape:
+            is_binary_logits = y_pred.shape[1] == 1 and not self.use_softmax
+            if not self.to_onehot_y and not is_binary_logits:
+                raise ValueError(f"ground truth has different shape ({y_true.shape}) from input ({y_pred.shape})")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)